jax/_src/interpreters/mlir.py

```diff
@@ -1339,21 +1361,46 @@ def get_lowering(primitive: core.Primitive) -> LoweringRule | None:
     loc = _source_info_to_location(eqn.primitive, eqn.params, source_info,
                                    ctx.name_stack)
     with source_info_util.user_context(eqn.source_info.traceback), loc:
-      override_rule = get_lowering(eqn.primitive)
-      if override_rule is not None:
-        rule = override_rule
-      elif eqn.primitive in _platform_specific_lowerings[ctx.platform]:
-        rule = _platform_specific_lowerings[ctx.platform][eqn.primitive]
-      elif eqn.primitive in xla._backend_specific_translations[ctx.platform]:
-        rule = xla_fallback_lowering(eqn.primitive)
-      elif eqn.primitive in _lowerings:
-        rule = _lowerings[eqn.primitive]
-      elif eqn.primitive in xla._translations:
-        rule = xla_fallback_lowering(eqn.primitive)
+      override_rule = get_override_lowering_rule(eqn.primitive)
+      if not ctx.lowering_parameters.multi_platform:
+        # Classic, single-platform lowering
+        # TODO(necula): unify the code paths when multi-platform is finished
+        if override_rule is not None:
+          rule = override_rule
+        elif eqn.primitive in _platform_specific_lowerings[ctx.platform]:
+          rule = _platform_specific_lowerings[ctx.platform][eqn.primitive]
+        elif eqn.primitive in xla._backend_specific_translations[ctx.platform]:
+          rule = xla_fallback_lowering(eqn.primitive)
+        elif eqn.primitive in _lowerings:
+          rule = _lowerings[eqn.primitive]
+        elif eqn.primitive in xla._translations:
+          rule = xla_fallback_lowering(eqn.primitive)
+        else:
+          raise NotImplementedError(
+              f"MLIR translation rule for primitive '{eqn.primitive.name}' not "
+              f"found for platform {ctx.platform}")
       else:
-        raise NotImplementedError(
-            f"MLIR translation rule for primitive '{eqn.primitive.name}' not "
-            f"found for platform {ctx.platform}")
+        rules: Sequence[tuple[Optional[Sequence[str]], Callable]]
```

Comment:
I defined MultiPlatformLoweringRule. I am leaving the type of the rule unspecified, because the actual type is too complicated, and I am not sure it is helpful for type checking.